flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/ProjectionTranslator.scala

```diff
@@ -327,4 +332,56 @@ object ProjectionTranslator {
       }
   }
 
+  /**
+    * Find and replace UDAGG function Call to UDAGGFunctionCall
+    *
+    * @param field    the expression to check
+    * @param tableEnv the TableEnvironment
+    * @return an expression with correct UDAGGFunctionCall type for UDAGG functions
+    */
+  def replaceUDAGGFunctionCall(field: Expression, tableEnv: TableEnvironment): Expression = {
```

Comment:
Couldn't we do the replacement during `replaceAggregationsAndProperties`? Save some duplicate code? Actually, `LogicalNode#resolveExpressions` is responsible for looking up calls. Maybe this logic has to be adapted a little bit.